Client/Frontend/Home/RecentTabsPanel.swift

```diff
@@ -0,0 +1,162 @@
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanelSiteTableViewController.notificationReceived(_:)), name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanelSiteTableViewController.notificationReceived(_:)), name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.accessibilityIdentifier = "Recently Closed Tabs List"
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    func notificationReceived(notification: NSNotification) {
+        switch notification.name {
+        case NotificationPrivateDataClearedHistory:
+            profile.prefs.setObject(nil, forKey: "recentlyClosedTabs")
+            break
+        case NotificationDynamicFontChanged:
+            reloadData()
+            break
+        default:
+            // no need to do anything at all
+            log.warning("Received unexpected notification \(notification.name)")
+            break
+        }
+    }
+
+    override func reloadData() {
+        if let recentTabs = profile.prefs.dictionaryForKey("recentlyClosedTabs") {
+            recentlyClosedTabs = recentTabs as! [String:[String]]
+        }
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
+
+        if let cell = cell as? TwoLineTableViewCell {
+            cell.setLines(recentlyClosedTabs["titles"]![indexPath.row], detailText: recentlyClosedTabs["urls"]![indexPath.row])
+            cell.imageView?.sd_setImageWithURL((recentlyClosedTabs["faviconUrls"]![indexPath.row] ).asURL!) { (img, err, type, url) -> Void in
+                guard img != nil else {
+                    cell.imageView?.setIcon(nil, withPlaceholder: FaviconFetcher.getDefaultFavicon(NSURL(string: (self.recentlyClosedTabs["urls"]![indexPath.row]))!))
+                    return
+                }
+            }
+        }
+
+        return cell
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        if let url = NSURL(string: (recentlyClosedTabs["urls"]![indexPath.row])) {
+            let visitType = VisitType.Typed    // Means History, too.
+            if let homePanelDelegate = homePanelDelegate,
+                historyPanel = historyPanel {
+                homePanelDelegate.homePanel(historyPanel, didSelectURL: url, visitType: visitType)
```

Comment:
This looks like the only place you're using `historyPanel`. Why not just pass `self` instead (since `RecentTabsPanel` is a `HomePanel`), and remove the awkward panel->panel dependency?